lib/schema/schema-to-drizzle.js

```diff
@@ -0,0 +1,302 @@
+@template {TSchemaName extends MapeoDoc['schemaName'] ? MapeoDocMap[TSchemaName] : any} TObjectType
+@param {TSchema} schema
+@returns {SchemaToDrizzleColumns<TSchema, TObjectType>}
+ */
+export function jsonSchemaToDrizzleColumns(schema) {
+  if (schema.type !== 'object' || !schema.properties) {
+    throw new Error('Cannot process JSONSchema as SQL table')
+  }
+  /** @type {Record<string, import('drizzle-orm/sqlite-core').AnySQLiteColumnBuilder>} */
+  const columns = {}
+  for (const [key, value] of Object.entries(schema.properties)) {
+    if (typeof value !== 'object') continue
+    if (isArray(value.type) || typeof value.type === 'undefined') {
+      throw new Error('Cannot process JSONSchema as SQL table')
+    }
+    switch (value.type) {
+      case 'boolean':
+        columns[key] = integer(key, { mode: 'boolean' })
+        break
+      case 'number':
+      case 'integer':
+        columns[key] = real(key)
+        break
+      case 'string': {
+        const enumValue = isStringArray(value.enum)
+          ? value.enum
+          : typeof value.const === 'string'
+          ? /** @type {[typeof value.const]} */ ([value.const])
+          : undefined
+        columns[key] = text(key, { enum: enumValue })
+        if (key === 'docId') {
+          columns[key] = columns[key].primaryKey()
+        }
+        break
+      }
+      case 'array':
+      case 'object':
+        columns[key] = customJson(key)
+        break
+      case 'null':
+        // Skip handling this right now
+        continue
+      default: {
+        /** @type {never} */
+        // eslint-disable-next-line no-unused-vars
+        const _exhaustiveCheck = value.type
+        continue
+      }
+    }
+    const defaultValue = getDefault(value)
+    if (typeof defaultValue !== 'undefined') {
+      columns[key] = columns[key].default(defaultValue)
+    }
+    if (isRequired(schema, key)) {
+      columns[key] = columns[key].notNull()
+    }
+  }
+  // Not yet in @mapeo/schema
+  columns.forks = customJson('forks').notNull()
+  return /** @type {any} */ (columns)
+}
+
+/**
+ * @template {JSONSchema7} T
+ * @param {T} value
+ * @returns {T['default']}
+ */
+function getDefault(value) {
+  return value.default
+}
+
+/**
+ * @param {JSONSchema7WithProps} schema
+ * @param {string} key
+ * @returns {boolean}
+ */
+function isRequired(schema, key) {
+  if (!isArray(schema.required)) return false
+  return schema.required.includes(key)
+}
+
+/**
+ * Tests whether a value is an array.
+ * @param {any} value
+ * @returns {value is readonly unknown[]}
+ */
+function isArray(value) {
+  // See: https://github.com/microsoft/TypeScript/issues/17002
+  return Array.isArray(value)
+}
+
+/**
+ * Check if a value is an array of strings of length at least one
+ * @param {any} value
+ * @returns {value is [string, ...string[]]}
+ */
+function isStringArray(value) {
+  return (
+    isArray(value) &&
+    value.every((v) => typeof v === 'string') &&
```

Comment:
was going to suggest prematurely optimizing this using `Array.some`, but the length `value` will be small here like 99% of the time right?

Reply:
Yes, will be small e.g. max will be the number of enum values. Also this only runs once on app startup so it's not something that needs to be fast.